### DISCUSSION TITLE
How to package a cmake based project correctly

### DISCUSSION THREAD

**Discussion by qiuwei** [xmake-io/xmake#4145] [Q&A] [answered]
I am trying to contribute to xmake repo by packaging a cmake-based project https://github.com/open-license-manager/licensecc.
I could build and install it without any issues.
However, the project does not include a pc file.
So when I use add_packages to add licencecc as a dependency of my project, the linkdir for licencecc is not properly added to the build target.

How could I fix this? Should I fork the upstream project and provide a valid pc file?

**Comment by waruqi** [comment, marked as answer]
> However, the project does not include a pc file.
So when I use add_packages to add licencecc as a dependency of my project, the linkdir for licencecc is not properly added to the build target.

xmake will patch .pc file automatically.

and add_packages does not pc file.  you need add `add_requires("licencecc")`
